mongoengine/queryset/queryset.py

```diff
@@ -422,9 +422,22 @@ def count(self, with_limit_and_skip=True):
 
         if with_limit_and_skip and self._len is not None:
             return self._len
-        count = self._cursor.count(with_limit_and_skip=with_limit_and_skip)
+
+        options = {}
+
+        if with_limit_and_skip:
+            if self._limit is not None:
+                options["limit"] = self._limit
+            if self._skip is not None:
+                options["skip"] = self._skip
+        if self._hint not in (-1, None):
+            options["hint"] = self._hint
+
+        count = self._cursor.collection.count_documents(filter=self._query, **options)
```

Comment:
I fear this change might be hiding a big performance degradation.

If I understand correctly, what would've been a very quick operation previously `SomeDocument.objects.count()` will now result in a full collection scan. Is that right? And if so, is that a concern for us? Should we conditionally use `estimated_document_count` if the query is empty[^0]?

[^0]: Keep in mind, I don't remember if an empty `_query` truly means a lack of filtering, or if we also have to look at `_initial_query`, document inheritance, etc.